lib-extra/src/main/java/com/diffplug/spotless/extra/java/EclipseFormatterStep.java

```diff
@@ -47,14 +40,14 @@ private EclipseFormatterStep() {}
 	private static final String FORMATTER_METHOD = "format";
 
 	/** Creates a formatter step for the given version and settings file. */
-	public static FormatterStep create(File settingsFile, Provisioner provisioner) {
-		return create(defaultVersion(), settingsFile, provisioner);
+	public static FormatterStep create(Iterable<File> settingsFiles, Provisioner provisioner) {
```

Comment:
I principally agree, lib-extra is the supplier, plugin-gradle/plugin-maven are the users. Contract should not be broken.
But the basic idea is still to have a common interface and allow configuration injection by reusing files. So here should not be a user of the interface, except the tests, which should by ignored for the interface design.
So the binary compat is a reason to keep the old interface and mark it as deprecated. But I would then still refactor the tests.